src/main/java/testapp/core/User.java

```diff
@@ -1,30 +1,40 @@
 package testapp.core;
 
-public class User {
-    private int id;
+import com.google.appengine.api.datastore.Entity;
+
+import java.io.Serializable;
+
+public class User implements Serializable {
+    private long ucn;
     private String name;
 
     public User() {
     }
 
-    public User(int id, String name) {
-        this.id = id;
+    public User(long ucn, String name) {
+        this.ucn = ucn;
         this.name = name;
     }
 
-    public int getId() {
-        return id;
-    }
-
-    public void setId(int id) {
-        this.id = id;
+    public long getUcn() {
+        return ucn;
     }
 
     public String getName() {
         return name;
     }
 
-    public void setName(String name) {
-        this.name = name;
+    public Entity toEntity(String kind) {
+        Entity entity = new Entity(kind);
+        entity.setProperty("ucn", this.ucn);
+        entity.setProperty("name", this.name);
+        return entity;
+    }
+
+    public static User fromEntity(Entity entity) {
+        Long id = (Long) entity.getProperty("ucn");
```

Comment:
I can't see how you are going to return the user's id, so it can be stored in the UserAccount